backend/courses/views.py

```diff
@@ -376,3 +384,122 @@ def get_queryset(self):
+        sender = request.user
+        recipient = get_object_or_404(User, id=request.recipient_id)
+
+        if not recipient:
+            res["message"] = "Recipient does not exist."
+            return JsonResponse(res, status=400)
+
+        # if friend request already exists but was rejected, we can reinstate it:
+        friend_req = self.queryset.get(sender=sender, recipient=recipient, status=Friendship.FriendshipStatus.REJECTED)
+        if not friend_req:
+            friend_req = self.queryset.get(sender=recipient, recipient=sender, status=Friendship.FriendshipStatus.REJECTED)
+        if friend_req:
+            friend_req.status = Friendship.FriendshipStatus.SENT
+            friendship.save()
+            res = model_to_dict(friendship)
+            res["message"] = "Friendship request sent successfully."
+        
+        if self.queryset.get(sender=sender, recipient=recipient, status=Friendship.FriendshipStatus.SENT).exists():
+            res["message"] = "Friendship request already exists."
+            return JsonResponse(res, status=status.HTTP_409_CONFLICT)
+
+        # if a request exists in the other direction, then we accept the friend request
+        friendship_request = self.queryset.get(sender=recipient, recipient=sender, status=Friendship.FriendshipStatus.SENT)
+        if friendship_request:
+            # accept friend request
+            friendship_request.setStatus("A")
+            friendship.save()
+            res = model_to_dict(friendship_request)
+            res["message"] = "Friendship request accepted."
+        else:
+            # create the friendship request
+            friendship = Friendship(sender=sender, recipient=recipient)
+            friendship.status = Friendship.FriendshipStatus.SENT
+            friendship.save()
+            res = model_to_dict(friendship)
+            res["message"] = "Friendship request sent successfully."
+
+        return JsonResponse(res, status=200)
+
+    def delete(self, request):
+        # either deletes a friendship or cancels a friendship request (depends on status)
+        res = {}
+        sender = request.user
+        recipient = get_object_or_404(User, id=request.recipient_id)
+
+        if not recipient:
+            res["message"] = "Recipient does not exist."
+            return JsonResponse(res, status=400)
+
+        # delete the friendship/request (see if it exists the other way around)
+        friendship = self.queryset.get(sender=sender, recipient=recipient)
+        if not friendship:
+            friendship = self.queryset.get(sender=recipient, recipient=sender)
+        if not friendship:
+            res["message"] = "Friendship/friendship request does not exist."
+        else:
+            friendship.delete()
+            res["message"] = "Friendship request removed."
+        return JsonResponse(res, status=200)
+    
+    # not sure what URL this method gets assigned to???
+    def handle(self, request):
```

Comment:
We can just handle this in the `POST` method - if user_a POSTs with friend=user_b, that should create the friendship with SENT status. Then if user_b POSTs with friend=user_a, that should find the existing friendship and change the status to ACCEPTED.